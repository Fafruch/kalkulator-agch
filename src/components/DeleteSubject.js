/**
 * Created by Fafruch on 18/07/2017.
 */

import React from 'react'
import PropTypes from 'prop-types'

const DeleteSubject = ({ id, onDelete }) => (
  <img
    src='https://png.icons8.com/delete-bin/ios7/100'
    alt='UsuÅ„ przedmiot'
    style={{ width: 32, height: 32 }}
    onClick={() => onDelete(id)}
  />
)
DeleteSubject.propTypes = {
  id: PropTypes.number.isRequired,
  onDelete: PropTypes.func.isRequired
}

export default DeleteSubject
