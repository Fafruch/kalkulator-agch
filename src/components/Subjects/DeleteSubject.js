import React from 'react'
import PropTypes from 'prop-types'

const DeleteSubject = ({ id, subjectsType, onDelete }) => (
  <img
    src='https://png.icons8.com/delete-bin/ios7/100'
    alt='UsuÅ„ przedmiot'
    onClick={() => onDelete(id, subjectsType)}
    className='subject-icon'
  />
)
DeleteSubject.propTypes = {
  id: PropTypes.string.isRequired,
  subjectsType: PropTypes.string.isRequired,
  onDelete: PropTypes.func.isRequired
}

export default DeleteSubject
