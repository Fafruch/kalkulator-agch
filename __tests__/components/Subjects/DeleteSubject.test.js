import React from 'react'
import { shallow } from 'enzyme'

import DeleteSubject from '../../../src/components/Subjects/DeleteSubject'

const setup = (setupProps = {}) => {
  const defaultProps = {
    id: '0',
    subjectsType: 'primary',
    onDelete: jest.fn()
  }
  const props = { ...defaultProps, ...setupProps }
  const wrapper = shallow(<DeleteSubject {...props} />)

  return {
    props,
    wrapper
  }
}

describe('DeleteSubject component', () => {
  test('should render', () => {
    const { wrapper } = setup()

    expect(wrapper).toMatchSnapshot()
  })

  test('should call onDelete when clicked on a button', () => {
    const { props, wrapper } = setup()

    wrapper.simulate('click')
    expect(props.onDelete).toBeCalled()
  })
})
