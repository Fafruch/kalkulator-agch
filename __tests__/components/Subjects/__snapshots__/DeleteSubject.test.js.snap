// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteSubject component should render 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <img
    alt="Usuń przedmiot"
    className="subject-icon"
    onClick={[Function]}
    src="https://png.icons8.com/delete-bin/ios7/100"
/>,
  "nodes": Array [
    <img
      alt="Usuń przedmiot"
      className="subject-icon"
      onClick={[Function]}
      src="https://png.icons8.com/delete-bin/ios7/100"
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <DeleteSubject
        id="0"
        onDelete={[Function]}
        subjectsType="primary"
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "id": "0",
          "onDelete": [Function],
          "subjectsType": "primary",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <img
          alt="Usuń przedmiot"
          className="subject-icon"
          onClick={[Function]}
          src="https://png.icons8.com/delete-bin/ios7/100"
/>,
        "_debugID": 2,
        "_renderedOutput": <img
          alt="Usuń przedmiot"
          className="subject-icon"
          onClick={[Function]}
          src="https://png.icons8.com/delete-bin/ios7/100"
/>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <DeleteSubject
    id="0"
    onDelete={[Function]}
    subjectsType="primary"
/>,
}
`;
